### ISSUE TITLE
feature: `cosign sign-blob` should accept `--certificate` and `--certificate-chain`

### ISSUE THREAD

**Issue by nsmith5** [sigstore/cosign#2635]
**Description**

While `cosign verify-blob` will accept a certificate and CA chain to verify against, it appears the signing command won't accept them. Feels reasonable to make these two commands symmetric.

**Comment by haydentherapper** [comment]
What would those be used for? The purpose of including those flags for “cosign sign” is to attach them to the OCI image. For sign-blob, there is nothing to attach to.

The use case I’m not sure if we support now would be uploading a certificate associated with a signing key to Rekor.

**Comment by nsmith5** [comment]
They end up working together with `--bundle` so that they get attached to that bundle if specified. This makes `cosign verify-blob --bundle bundle.txt --certificate-chain chain.pem blob.txt` work when you byo CA

**Comment by haydentherapper** [comment]
SG, I would also test for how this interacts with Rekor. Though this is likely related to the other issue you filed regarding precedence between keys and certs.

**Comment by nsmith5** [comment]
Yeah this was kind of a yak shave moment because I went to go test `cosign verify-blob` for that other issue and figured out I couldn't sign 😆

**Comment by znewman01** [comment]
See also https://github.com/sigstore/cosign/issues/2511
